docs/mp/openapi/01_openapi.adoc

```diff
@@ -182,15 +160,241 @@ include::{common-openapi-doc}[common-config]
 Assign these settings in `META-INF/microprofile-config.properties`.
 
 
-== Accessing the OpenAPI document
-Now your Helidon MP application will automatically respond to an additional endpoint --
- `/openapi` -- and it will return the OpenAPI document describing the endpoints
+== REST Endpoint
+
+Helidon MP application will automatically respond to built-in endpoint --
+`/openapi` -- and it will return the OpenAPI document describing the endpoints
 in your application.
 
-By default, per the MicroProfile OpenAPI spec, the default format of the OpenAPI document is YAML. 
+By default, per the MicroProfile OpenAPI spec, the default format of the OpenAPI document is YAML.
 There is not yet an adopted IANA YAML media type, but a proposed one specifically
 for OpenAPI documents that has some support is `application/vnd.oai.openapi`.
 That is what Helidon returns, by default.
 
-In addition a client can specify the HTTP header `Accept:` as either `application/vnd.oai.openapi+json` or `application/json`
+In addition, a client can specify the HTTP header `Accept:` as either `application/vnd.oai.openapi+json` or `application/json`
 to request JSON. Alternatively, the client can pass the query parameter `format` as either `JSON` or `YAML` to receive `application/json` or `application/vnd.oai.openapi` (YAML) output, respectively.
+
+== Configuration
+
+Helidon OpenAPI configuration is supported by the following class:
+
+include::../../shared/config/io.helidon.openapi.SEOpenAPISupport.adoc[]
+
+=== Helidon MP OpenAPI configuration.
+
+include::../../shared/config/io.helidon.openapi.internal.OpenAPIConfigImpl.Builder.adoc[]
+
+=== Smallrye Configuration
+
+Helidon MP OpenAPI implementation is based on link:https://smallrye.io/[Smallrye]. The configuration is backed by the following class:
+include::../../shared/config/io.smallrye.openapi.api.OpenApiConfig.adoc[]
+
+=== Configuration Example
+
+The classes with filters should be specified in `META-INF/microprofile-config.properties`:
+
+.Example Configuration
+[source,properties]
+----
+mp.openapi.filter=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIFilter
+mp.openapi.model.reader=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIModelReader
+----
+
```

Comment:
Remove these lines. The config example is also later and I think things flow better if we introduce the Java code first and then show the config that's needed for the Java classes to be used.

Reply:
Done)